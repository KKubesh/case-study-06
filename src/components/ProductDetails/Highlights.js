import React, {Component} from 'react';

class Highlights extends Component {
    state = {
        details: [
            "Wattage Output: 1100 Watts",
            "Number of Speeds: 3",
            "Capacity (volume): 72.0 Oz",
            "Appliance Capabilities: Blends",
            "Includes: Travel Lid",
            "Material: Plastic",
            "Finish: Painted",
            "Metal Finish: Chrome",
            "Safety and Security Features: Non-Slip Base",
            "Care and Cleaning: Easy-To-Clean, Dishwasher Safe Parts"
        ]
    }
    render () {
        let detailItems = this.state.details.map(item => {
            return (
                <li><span>{item}</span></li>
            )
        })
        return (
            <div className="product-highlights" data-testid="product-highlights">
                <h2 data-testid="highlights-title">product highlights</h2>
                <ul data-testid="highlights-details">
                    {detailItems}
                </ul>
            </div>
        )
    }
}

export default Highlights;