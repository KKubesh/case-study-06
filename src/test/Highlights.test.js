import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Highlights from '../components/ProductDetails/Highlights';

afterEach(cleanup);

test('<Highlights />', () => {
    const { queryByTestId, container } = render(<Highlights/>);
    expect(queryByTestId('highlights-title').textContent).toBe('product highlights');
    expect(queryByTestId('highlights-title')).toBeTruthy();
    expect(queryByTestId('highlights-details')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
});