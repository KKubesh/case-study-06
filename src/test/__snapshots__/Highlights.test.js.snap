// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlights /> 1`] = `
<div
  class="product-highlights"
  data-testid="product-highlights"
>
  <h2
    data-testid="highlights-title"
  >
    product highlights
  </h2>
  <ul
    data-testid="highlights-details"
  >
    <li>
      Wattage Output: 1100 Watts
    </li>
    <li>
      Number of Speeds: 3
    </li>
    <li>
      Capacity (volume): 72.0 Oz
    </li>
    <li>
      Appliance Capabilities: Blends
    </li>
    <li>
      Includes: Travel Lid
    </li>
    <li>
      Material: Plastic
    </li>
    <li>
      Finish: Painted
    </li>
    <li>
      Metal Finish: Chrome
    </li>
    <li>
      Safety and Security Features: Non-Slip Base
    </li>
    <li>
      Care and Cleaning: Easy-To-Clean, Dishwasher Safe Parts
    </li>
  </ul>
</div>
`;
